python/ray/tune/integration/lightgbm.py

```diff
@@ -14,17 +23,24 @@ class TuneCallback:
     pass
 
 
-class TuneReportCallback(TuneCallback):
-    """Create a callback that reports metrics to Ray Tune.
+class TuneReportCheckpointCallback(TuneCallback):
+    """Creates a callback that reports metrics and checkpoints model.
+
+    Saves checkpoints after each validation step. Also reports metrics to Tune,
+    which is needed for checkpoint registration.
 
     Args:
         metrics: Metrics to report to Tune. If this is a list,
             each item describes the metric key reported to LightGBM,
             and it will reported under the same name to Tune. If this is a
             dict, each key will be the name reported to Tune and the respective
-            value will be the metric key reported to LightGBM. If this is None,
+            value will be the metric key reported to LightGBM.
+        filename: Filename of the checkpoint within the checkpoint
+            directory. Defaults to "checkpoint". If this is None,
             all metrics will be reported to Tune under their default names as
             obtained from LightGBM.
+        frequency: How often to save checkpoints. Per default, a
+            checkpoint is saved every five iterations.
```

Comment:
should we set default = 0?

Reply:
checkpoint_at_end will already save one at the end by default. then, extra checkpointing can be added w/ this param.

Reply:
Agreed - updated to 0 per default